EIPS/eip-draft_transaction_representation.md

```diff
@@ -0,0 +1,67 @@
+---
+eip: 7023
+title: Automatic Transaction Representation v1
+description: Guidelines for automatically generating a human-readable representation of a transaction and its corresponding EIP-712 signature.
+author: Gavin John (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/automatic-transaction-representation/14256
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-12
+requires: 20, 712, 721, 1046, 1155
+---
+
+## Abstract
+
+This EIP proposes an extendable standard to allow wallets to provide a rich representation of a transaction or signature request to the user. This representation is generated automatically from the transaction data and/or signature request.
+
+## Motivation
+
+While signing transactions is made much more secure by [EIP-712](./eip-712.md), most wallets end up displaying the raw JSON/YAML representation of the transaction to the user. This is not very user friendly and can lead to users being tricked into signing transactions that they did not intend to sign, as the average user is unable to read it.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Only Final EIPs may be referenced in this EIP or future versions of 'Automatic Transaction Representation' EIPs. Wallets SHOULD refer to 'Automatic Transaction Representation' EIPs by their full title, including the version. 'Automatic Transaction Representation' EIPs SHOULD include the full list of rules, as well as indicating the changes made in the Rationale section.
```

Comment:
You can't really place restrictions on how EIPs will be written. If someone comes along with a v2 of this proposal, they can refer to whatever statuses they like.

```suggestion
Wallets SHOULD refer to 'Automatic Transaction Representation' EIPs by their full title, including the version.
```

Reply:
Is there any rule in EIP-1 that says that an EIPs can't place additional restrictions on other EIPs that choose to follow it?